src/clj/athens/self_hosted/components/web.clj

```diff
@@ -20,13 +20,10 @@
 
 (defn close-handler
   [channel status]
-  (let [username (clients/get-client-username channel)
-        ;; TODO: max-tx shouldn't be 42
-        presence-offline-event (athens.common-events/build-presence-offline-event 42 username)]
+  (let [username (clients/get-client-username channel)]
+    (presence/goodbye-handler channel)
     (clients/remove-client! channel)
```

Comment:
We might want to `remove-client!` before calling `goodbye-handler` (because it's supposed to broadcast to all connected clients)

Reply:
Sgtm, done.